webapp/packages/plugin-connections/src/ConnectionAuthService.ts

```diff
@@ -87,11 +88,14 @@ export class ConnectionAuthService extends Dependency {
       .filter(handler => handler.enabled && (!handler.savePassword || resetCredentials))
       .map(handler => handler.id);
 
-    if (connection.authNeeded || resetCredentials || networkHandlers.length > 0) {
+    if (connection.authNeeded || (resetCredentials && networkHandlers.length > 0)) {
       await this.commonDialogService.open(DatabaseAuthDialog, {
         connection: key,
         networkHandlers,
       });
+      if (resetCredentials && isConnectedInitially) {
+        await this.connectionInfoResource.init(key);
+      }
```

Comment:
can you please describe the case?
`DatabaseAuthDialog` will call `init` on click on `Login` button, why it's double called?
Why you joined networkHandlers and reset credentials flags?